### PR TITLE
Add a check for blank search term before searching

### LINKED ISSUE #769: Disable homepage search submission without a query
## Description

Currently, the homepage allows users to submit a search with no search query. We should block that, as it's one of the only explicit user-facing paths to the raw, query-less /search/ endpoint which has been identified as a problem in #596 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #769 by @zackkrida

## Description

This PR adds a check for search term before submitting a query from the search page. This prevents the user from getting to the blank `/search/` url.

## Testing Instructions

Go to the homepage and click the search button. You should not be redirected, or see any changes.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pages/index.vue [hunk 1]
```diff
@@ -201,8 +201,9 @@ const HomePage = {
 
     const searchTerm = ref('')
     const handleSearch = async () => {
+      if (!searchTerm.value) return
```
Comment (sarayourfriend) [inline src/pages/index.vue:204]: I wonder if we should `.trim()` the search term as well? Like if it's all spaces, do we still want to submit it?

DuckDuckGo actually displays an error when you try to submit a string of spaces.

Captura de Tela 2022-02-08 às 05 32 18

**Review by sarayourfriend — APPROVED** [review]
I was checking out how DuckDuckGo and Google handle this, just to make sure we match in behavior and because I was wondering if it was worth adding a `minlength="1"` to the input.

DuckDuckGo allows you to submit the form but automatically redirects you back to their homepage.

Google ignores the search request if JavaScript is enabled. If you disabled JavaScript, they also just redirect you back to the homepage if you submit an empty query.

Given this PR makes our implementation match Google's homepage, it seems like that's probably safe to do. They also do _not_ include anything like `minlength="1"` on the form.

We also don't have a way of handling JavaScript-disabled clients currently anyway so the second version isn't as important.

I had one suggestion but take it or leave it, not a blocker! LGTM 😁

File: src/pages/index.vue (hunk 1 above)
Comment (obulat) [reply src/pages/index.vue:204]: This issue is actually already handled by Vue with the `v-model`'s `.trim` modifier:
https://github.com/WordPress/openverse-frontend/blob/31ba45713247e31c9ae2c7bd251808d586b0a6e9/src/pages/index.vue#L45
I did go back and test it, and it does work, you can't search for ' '.

Comment (sarayourfriend) [reply src/pages/index.vue:204]: 😱 I should have tried it out before commenting!

**Comment by obulat** [comment]
Very interesting investigations, @sarayourfriend !

> I was wondering if it was worth adding a `minlength="1"` to the input.

I also had this thought in the back of my head, so thank you for sharing your investigation results.

**Review by krysal — APPROVED** [review]
Nice!

**Pull request merged**